### PR TITLE
Move AsyncAwait helpers into NIOCore

### PR DESCRIPTION
**Pull request by fabianfett**
### Motivation

With the release of Swift 5.5 we should move our AsyncAwait helpers into our stable API.

### Changes

- Moved the `AsyncAwaitSupport.swift` to `NIOCore`
- Moved the `NIOCore.System.hasAsyncAwaitSupport` into the `AsyncAwaitSupport.swift` file

### Open questions

- Do we want to move those extensions next to their objects?

### Result

Async/await API is declared stable.

### CODE REVIEW

File: Sources/NIOCore/AsyncAwaitSupport.swift [hunk 1]
```diff
@@ -215,3 +213,15 @@ extension ChannelPipeline {
...
 #endif
+
+#if compiler(>=5.5)
+fileprivate let hasAsyncAwait = true
+#else
+fileprivate let hasAsyncAwait = false
+#endif
+
+extension NIOCore.System {
+    public static var hasAsyncAwaitSupport: Bool {
```
Comment (Lukasa) [inline Sources/NIOCore/AsyncAwaitSupport.swift:224]: Do we need to keep this?

**Comment by Lukasa** [comment]
Note that nightly is failing due to missing sendable conformances. We need to add this, but it's not blocking us from moving forward for now.

File: Sources/NIOCore/AsyncAwaitSupport.swift (hunk 1 above)
Comment (fabianfett) [reply Sources/NIOCore/AsyncAwaitSupport.swift]: Removed

**Review by Lukasa — APPROVED** [review]

**Comment by fabianfett** [comment]
`swift:nightly-main` still falling because of `Sendable` requirements.

**Pull request merged**